app/integrations/discord/interactions/resolvers/ping.rb

```diff
@@ -0,0 +1,28 @@
+module Discord
+  module Interactions
+    module Resolvers
+      class Ping < Resolver
+        EXPECTED_GLOBAL_NAME = 'Discord'.freeze
+        PONG_RESPONSE_TYPE = 1
+
+        Response = Struct.new(:type)
+
+        def call
+          @response_status = RESPONSE_STATUS[:UNAUTHORIZED]
+          @body = {}
+
+          return unless valid_signature? && correct_global_name?
+
+          @response_status = RESPONSE_STATUS[:OK]
+          @body = Response.new(PONG_RESPONSE_TYPE)
+        end
```

Comment:
Let's follow the pattern we defined for the messages service classes, and let's raise ruby errors to handle unhappy paths.

```suggestion
        def call
          raise InvalidSignatureError unless valid_signature? && correct_global_name?
          @type = PONG_RESPONSE_TYPE
        end
```

Then in the controller we can rescue these errors and respond with the corresponding status code and body.

```rb
post '/interactions' do
  # ...
rescue InvalidSignatureError
  status 401
end
```

That way, as I said above, we can also leave that response status knowledge to the controller. What do you think?

Reply:
I agree:

```ruby
post '/interactions' do
  # ...
  status :ok
rescue InvalidSignatureError
  status :unauthorized
  { error: 'unauthorized request' }
end
```